hilti/runtime/include/types/stream.h

```diff
@@ -856,13 +856,7 @@ class UnsafeConstIterator {
     }
 
     Byte _dereference() const {
-        // TODO(bbannier): Ideally we would `assert(_chunk)` here so clang-tidy
-        // knows that `_chunk` is always not null. Unfortunately it looks like
-        // that fails to it pruning that CFG edge so we have to return instead
-        // here. This should be inconsequential to users as this function must
-        // not be called if the data is invalid.
-        if ( ! _chunk )
-            internalError("dereference of invalid iterator");
+        assert(_chunk);
```

Comment:
This was the main motivation for this patch since this function can get pretty hot for realistic workloads. By replacing the branch with an `assert` we can make the code less branchy and potentially also easier to optimize (viewed from the optimizer `assert` reduces the possible states). Overall the changes in this branch shave off about 1%.